### ISSUE TITLE
[P2] Migrate grpc client async calls to use thread_pool

### ISSUE THREAD

**Issue by kcenon** [kcenon/network_system#278]
## Summary

Replace `std::thread().detach()` in gRPC client async calls with thread_pool task submission.

## Current Implementation

**File**: `src/protocols/grpc/client.cpp`

```cpp
// Lines 664-671
auto call_raw_async(const std::string& method,
                    const std::vector<uint8_t>& request,
                    std::function<void(Result<grpc_message>)> callback,
                    const call_options& options) -> void
{
    // Problem: Creates detached thread for each async call
    std::thread([this, method, request, callback, options]() {
        auto result = call_raw(method, request, options);
        if (callback) {
            callback(std::move(result));
        }
    }).detach();
}
```

## Problem

- Creates a detached thread for every async gRPC call
- No control over thread lifecycle
- Potential thread explosion under high load
- No way to cancel pending async operations

### Use thread_pool for async execution

```cpp
#include "kcenon/network/integration/thread_integration.h"

auto call_raw_async(const std::string& method,
                    const std::vector<uint8_t>& request,
                    std::function<void(Result<grpc_message>)> callback,
                    const call_options& options) -> void
{
    // Submit to thread pool instead of creating detached thread
    integration::thread_integration_manager::instance().submit_task(
        [this, method, request, callback, options]() {
            auto result = call_raw(method, request, options);
            if (callback) {
                callback(std::move(result));
            }
        });
}
```

### Optional: Add cancellation support

```cpp
#include <kcenon/thread/core/cancellation_token.h>

auto call_raw_async(const std::string& method,
                    const std::vector<uint8_t>& request,
                    std::function<void(Result<grpc_message>)> callback,
                    const call_options& options,
                    kcenon::thread::cancellation_token token = {}) -> void
{
    integration::thread_integration_manager::instance().submit_task(
        [this, method, request, callback, options, token]() {
            if (token.is_cancelled()) {
                if (callback) {
                    callback(error<grpc_message>(..., "Cancelled"));
                }
                return;
            }
            auto result = call_raw(method, request, options);
            if (callback) {
                callback(std::move(result));
            }
        });
}
```

## thread_system Features to Use

- `thread_integration_manager`: Centralized thread pool access
- `thread_pool`: Managed task execution
- `cancellation_token`: Optional cancellation support

## Tasks

- [ ] Replace `std::thread().detach()` with thread_pool submission
- [ ] Consider adding cancellation token support
- [ ] Add error handling for pool submission failure
- [ ] Add unit tests
- [ ] Performance test under high load

## Acceptance Criteria

- [ ] No `std::thread().detach()` in gRPC client
- [ ] Async calls submitted to thread pool
- [ ] All existing tests pass
- [ ] No performance regression

## Related

- Epic: #271
- Depends on: #272
